server/src/storage/time_series.rs

```diff
@@ -0,0 +1,212 @@
+
+            write_batch_to_parquet(batch, path.clone());
+
+            Ok(())
+        }
+    }
+
+    /// Return the size in bytes of `segment`.
+    fn get_size_of_segment(segment: &RecordBatch) -> usize {
+        let mut total_size: usize = 0;
+
+        for column in segment.columns() {
+            // TODO: How is this calculated internally?
+            total_size += column.get_array_memory_size()
+        }
+
+        total_size
+    }
+
+    /// Return the record batch schema used for compressed segments.
+    fn get_compressed_segment_schema() -> Schema {
+        Schema::new(vec![
+            Field::new("model_type_id", DataType::UInt8, false),
+            Field::new("timestamps", DataType::Binary, false),
+            Field::new("start_time", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("end_time", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("values", DataType::Binary, false),
+            Field::new("min_value", ArrowValue::DATA_TYPE, false),
+            Field::new("max_value", ArrowValue::DATA_TYPE, false),
+            Field::new("error", DataType::Float32, false),
+        ])
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use datafusion::arrow::array::UInt8Array;
+    use datafusion::arrow::datatypes::DataType::UInt8;
+    use datafusion::arrow::datatypes::{Field, Schema};
+    use datafusion::arrow::record_batch::RecordBatch;
+
+    use crate::storage::time_series::{test_util, CompressedTimeSeries};
+
+    #[test]
+    fn test_can_append_valid_compressed_segment() {
+        let mut time_series = CompressedTimeSeries::new();
+        time_series.append_segment(test_util::get_compressed_segment_record_batch());
+
+        assert_eq!(time_series.compressed_segments.len(), 1)
+    }
+
+    #[test]
+    #[should_panic(expected = "Schema of record batch does not match compressed segment schema.")]
+    fn test_panic_if_appending_invalid_compressed_segment() {
+        let invalid = test_util::get_invalid_compressed_segment_record_batch();
+
+        let mut time_series = CompressedTimeSeries::new();
+        time_series.append_segment(invalid);
+    }
+
+    #[test]
+    fn test_compressed_time_series_size_updated_when_appending() {
+        let mut time_series = CompressedTimeSeries::new();
+        time_series.append_segment(test_util::get_compressed_segment_record_batch());
+
+        assert!(time_series.size_in_bytes > 0);
+    }
+
+    #[test]
+    fn test_cannot_save_empty_compressed_segments_to_parquet() {
+        let mut empty_time_series = CompressedTimeSeries::new();
+        let result = empty_time_series.save_to_parquet("key".to_owned());
+
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_get_size_of_segment() {
+        let segment = test_util::get_compressed_segment_record_batch();
+
+        assert_eq!(CompressedTimeSeries::get_size_of_segment(&segment), 2032);
```

Comment:
Maybe define a named constant in `test_util` with this information.

Reply:
Added.